newIDE/app/src/PropertiesEditor/PropertiesMapToSchema.js

```diff
@@ -114,6 +115,36 @@ export default (
         getLabel,
         getDescription,
       };
+    } else if (valueType === 'behavior') {
+      let choices = [];
+      if (object && property.getExtraInfo().size() > 0) {
+        const behaviorType = property.getExtraInfo().at(0);
+        choices = object
+          .getAllBehaviorNames()
+          .toJSArray()
+          .map(name =>
+            object.getBehavior(name).getTypeName() === behaviorType
+              ? name
+              : null
+          )
+          .filter(Boolean)
+          .map(value => ({ value, label: value }));
+      }
+      return {
+        name,
+        valueType: 'string',
+        getChoices: () => choices,
```

Comment:
Not 100% sure but if that is a function I think you should rebuild the `choices` array each time it is run instead of building it when making the Schema and always returning it?

Reply:
I tried to add a behavior to the object and it appeared in the combobox.
Did you smell another potential issue?

Reply:
My gut feeling would be indeed to load the choices lazily inside the callback - would ensure validity and freshness but not sure if there is a real problem if you don't do it.